[Bug]: MCP client panics when processing object schemas without properties field

**Environment (please complete the following):**
- OS: macOS 24.6.0 (Darwin ARM64)
- kubectl-ai version (run `kubectl-ai version`): dev (built from source)
- LLM provider: openai (via omnia.reainternal.net)
- LLM model: gpt-5-chat

**Describe the bug**
kubectl-ai panics with an interface conversion error when using `--mcp-client` flag and connecting to MCP servers that return object schemas without a `properties` field. The panic occurs during schema conversion in the MCP client code.

**To Reproduce**
Steps to reproduce the behavior:
1. Configure an MCP server that returns tool schemas with object types using `additionalProperties` but no `properties` field (e.g., `@gleanwork/local-mcp-server`)
2. Add the server to `~/.config/kubectl-ai/mcp.yaml`
3. Run command `kubectl-ai --mcp-client`
4. See error:
```
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]: github.com/GoogleCloudPlatform/kubectl-ai/pkg/mcp.convertMCPMapSchema({0x140005b3a88, 0x7}, 0x140002920c0) /Users/bharat.sardana/workspace/kubectl-ai/pkg/mcp/client.go:305 +0x810 github.com/GoogleCloudPlatform/kubectl-ai/pkg/mcp.convertMCPInputSchema(0x140004b6a20) /Users/bharat.sardana/workspace/kubectl-ai/pkg/mcp/client.go:249 +0x2fc github.com/GoogleCloudPlatform/kubectl-ai/pkg/mcp.convertMCPToolsToTools({0x140003a8288, 0x4, 0x1400032ed40?}) /Users/bharat.sardana/workspace/kubectl-ai/pkg/mcp/client.go:209 +0x130 github.com/GoogleCloudPlatform/kubectl-ai/pkg/mcp.listClientTools({0x106522b98?, 0x1400032ed40?}, 0x0?, {0x14000385580, 0x5}) /Users/bharat.sardana/workspace/kubectl-ai/pkg/mcp/client.go:440 +0x114
```

**Expected behavior**
kubectl-ai should gracefully handle object schemas that use `additionalProperties` without `properties` field, as this is valid JSON Schema. The MCP client should initialize successfully and make the tools available.

**Additional context**
- **Root cause**: The `convertMCPMapSchema` function in `pkg/mcp/client.go` assumes all object-type schemas have a `properties` field, but JSON Schema allows object schemas with only `additionalProperties`.
- **Problematic schema example** (from `@gleanwork/local-mcp-server`'s `people_profile_search` tool):
```json
{
  "type": "object",
  "additionalProperties": {"type": "string"},
  "propertyNames": {"enum": ["email", "first_name", "last_name", "manager_email", "department", "title", "location", "city", "country", "state", "region", "business_unit", "team", "team_id", "nickname", "preferred_name", "roletype", "reportsto", "startafter", "startbefore", "industry", "has", "from"]},
  "description": "Allowed facet fields: email, first_name, last_name, manager_email, department, title, location, city, country, state, region, business_unit, team, team_id, nickname, preferred_name, roletype, reportsto, startafter, startbefore, industry, has, from. Provide as { \"facet\": \"value\" }."
}
```
Specific line causing panic: `pkg/mcp/client.go:305` where `schemaMap["properties"].(map[string]interface{})` is called without checking if `properties` field exists
Analysis: Tested individual MCP servers and identified that the Glean server's people_profile_search tool has an object schema with missing `properties` field, which is valid JSON Schema but breaks kubectl-ai's conversion logic
**Workaround:** Temporarily disable the problematic MCP server in config
**Proposed fix location:** Add nil/existence checks in `convertMCPMapSchema` function before accessing `schemaMap["properties"]` to handle object schemas that rely on additionalProperties without explicit `properties` .

**MCP Configuration for reproduction:**
```yaml
Servers:
- Args:
  - -y
  - "@gleanwork/local-mcp-server"
  Auth: null
  Command: npx
  Env:
    GLEAN_INSTANCE: "your-instance"
    GLEAN_API_TOKEN: "your-token"
  Name: glean
  OAuthConfig: null
  Timeout: 0
  URL: ""
  UseStreaming: false
```

--- a/pkg/mcp/client.go
+++ b/pkg/mcp/client.go
@@ -245,7 +245,7 @@ func convertMCPInputSchema(mcpInputSchema *mcp.ToolInputSchema) (*gollm.Schema, 
 					return nil, fmt.Errorf("converting MCP input schema to tool input schema: %w", err)
 				}
 				gollmSchema.Properties[key] = gollmValue
-			} else if valueMap, ok := value.(map[string]interface{}); ok {
+			} else if valueMap, ok := value.(map[string]interface{}); ok && valueMap != nil {
 				gollmValue, err := convertMCPMapSchema(key, valueMap)
 				if err != nil {
 					return nil, fmt.Errorf("converting MCP input schema to tool input schema: %w", err)
@@ -261,6 +261,10 @@ func convertMCPInputSchema(mcpInputSchema *mcp.ToolInputSchema) (*gollm.Schema, 
 }
 
 func convertMCPMapSchema(key string, schemaMap map[string]interface{}) (*gollm.Schema, error) {
+	if schemaMap == nil {
+		return nil, fmt.Errorf("schema map is nil for key %q", key)
+	}
+
 	gollmSchema := &gollm.Schema{}
 
 	if descriptionObj, ok := schemaMap["description"]; ok {
@@ -302,12 +306,22 @@ func convertMCPMapSchema(key string, schemaMap map[string]interface{}) (*gollm.S
 	case "object":
 		gollmSchema.Type = gollm.TypeObject
 		gollmSchema.Properties = make(map[string]*gollm.Schema)
-		for key, value := range schemaMap["properties"].(map[string]interface{}) {
-			propertySchema, err := convertMCPMapSchema(key, value.(map[string]interface{}))
-			if err != nil {
-				return nil, fmt.Errorf("converting MCP input schema to tool input schema: %w", err)
+		if propertiesObj, ok := schemaMap["properties"]; ok && propertiesObj != nil {
+			properties, ok := propertiesObj.(map[string]interface{})
+			if !ok {
+				return nil, fmt.Errorf("properties field is not a map for key %q: %+v", key, schemaMap)
 			}
-			gollmSchema.Properties[key] = propertySchema
+			for key, value := range properties {
+				valueMap, ok := value.(map[string]interface{})
+				if !ok {
+					return nil, fmt.Errorf("property value is not a map for key %q: %+v", key, value)
+				}
+				propertySchema, err := convertMCPMapSchema(key, valueMap)
+				if err != nil {
+					return nil, fmt.Errorf("converting MCP input schema to tool input schema: %w", err)
+				}
+				gollmSchema.Properties[key] = propertySchema
+			}
 		}
 	default:
 		return nil, fmt.Errorf("unexpected input schema type %q for key %q: %+v", mcpType, key, schemaMap)

